app/models/reports/region_summary_schema.rb

```diff
@@ -296,9 +304,41 @@ def visited_without_bp_taken(with_ltfu: false)
       end
     end
 
-    memoize def diabetes_treatment_outcome_breakdown(blood_sugar_risk_state)
+    memoize def diabetes_treatment_outcome_breakdown_rates(blood_sugar_risk_state)
       region_period_cached_query(__method__, blood_sugar_risk_state: blood_sugar_risk_state) do |entry|
-        diabetes_treatment_outcome_breakdown_rates(entry, blood_sugar_risk_state)
+        diabetes_treatment_outcome_breakdown_rates_of_region(entry, blood_sugar_risk_state)
+      end
+    end
+
+    memoize def diabetes_treatment_outcome_breakdown_counts(blood_sugar_risk_state)
+      region_period_cached_query(__method__, blood_sugar_risk_state: blood_sugar_risk_state) do |entry|
+        diabetes_treatment_outcome_breakdown_counts_of_region(entry, blood_sugar_risk_state)
+      end
+    end
+
+    memoize def diabetes_patients_with_bs_taken_breakdown_counts
+      region_period_cached_query(__method__) do |entry|
+        bs_taken_breakdown_hash = {}
+        Reports::PatientBloodSugar.blood_sugar_risk_states.keys.each do |blood_sugar_risk_state|
+          BloodSugar.blood_sugar_types.keys.each do |blood_sugar_type|
+            bs_taken_breakdown_hash[[blood_sugar_risk_state.to_sym, blood_sugar_type.to_sym]] =
+              diabetes_under_care(blood_sugar_risk_state, blood_sugar_type)[entry.region.slug][entry.period]
+          end
+        end
+        bs_taken_breakdown_hash
+      end
+    end
+
+    memoize def diabetes_patients_with_bs_taken_breakdown_rates
+      region_period_cached_query(__method__) do |entry|
+        bs_taken_breakdown_hash = {}
+        Reports::PatientBloodSugar.blood_sugar_risk_states.keys.each do |blood_sugar_risk_state|
+          BloodSugar.blood_sugar_types.keys.each do |blood_sugar_type|
+            bs_taken_breakdown_hash[[blood_sugar_risk_state.to_sym, blood_sugar_type.to_sym]] =
+              diabetes_under_care(blood_sugar_risk_state, blood_sugar_type)[entry.region.slug][entry.period]
+          end
+        end
+        rounded_percentages(bs_taken_breakdown_hash)
```

Comment:
We weren't able to reuse `diabetes_patients_with_bs_taken_breakdown_counts` here trivially, because `region_period_cached_query` does some grouping and nesting of data for regions and periods.